src/SourceGenerators/Uno.UI.SourceGenerators/Helpers/SymbolExtensions.cs

```diff
@@ -371,33 +357,17 @@ public static IEnumerable<INamedTypeSymbol> GetAllInterfaces(this ITypeSymbol? s
 					yield return (INamedTypeSymbol)symbol;
 				}
 
-				do
+				foreach (var @interface in symbol.AllInterfaces)
 				{
-					foreach (var intf in symbol.Interfaces)
-					{
-						yield return intf;
-
-						foreach (var innerInterface in intf.GetAllInterfaces())
-						{
-							yield return innerInterface;
-						}
-					}
-
-					symbol = symbol?.BaseType;
-
-					if (symbol == null)
-					{
-						break;
-					}
-
-				} while (symbol.SpecialType != SpecialType.System_Object);
+					yield return @interface;
+				}
 			}
 		}
 
 		public static bool IsNullable(this ITypeSymbol type)
 		{
 			return ((type as INamedTypeSymbol)?.IsGenericType ?? false)
-				&& type.OriginalDefinition.ToDisplayString().Equals("System.Nullable<T>", StringComparison.OrdinalIgnoreCase);
+				&& type.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T;
```

Comment:
`ToDisplayString` is a bit slow. Comparing SpecialType is both more reliable and more efficient.